packages/typespec-azure-resource-manager/test/resource.test.ts

```diff
@@ -652,4 +652,68 @@ describe("typespec-azure-resource-manager: ARM resource model", () => {
     strictEqual(resources.length, 1);
     ok(resources[0].typespecType.properties.has("extendedLocation"));
   });
+
+  it("emits correct fixed union name parameter for resource", async () => {
+    const { program, diagnostics } = await checkFor(`
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Contoso;
+
+      @doc("Widget resource")
+      model Widget is ProxyResource<WidgetProperties> {
+         ...ResourceNameParameter<Widget, NameType=WidgetNameType>;
+      }
+
+      @doc("The properties of a widget")
+      model WidgetProperties {
+         size: int32;
+      }
+
+      /** different type of widget used on resource path */
```

Comment:
I feel like we should add an expected output test in typespec-autorest for each of these scenarios